### ISSUE TITLE
lein javac sometimes alters the :target-path, sometimes causing class resolution issues

### ISSUE THREAD

**Issue by djsweet** [technomancy/leiningen#2120]
In cases where `:target-path` is defined with a `"%s"` parameter, `lein javac` alters the `:target-path` by appending `"+xxxxxxxx"`, where `"xxxxxxxx"` is the first 8 characters of the SHA1 hash of the profile map (see [line 446 in `leiningen-core/src/leiningen/core/project.clj`](https://github.com/technomancy/leiningen/blob/2.6.1/leiningen-core/src/leiningen/core/project.clj#L446), I haven't been able to verify that the profile even gets merged properly when used this way). In some cases, the resulting `:target-path` being separate from the `:target-path` without `lein javac` causes class resolution issues (e.g., when writing an annotation processor that calls into Clojure, and applying it against the project still being compiled). Removing the `"%s"` from `:target-path` is a valid workaround.

I have [a patch for this](https://github.com/djsweet/leiningen/tree/javac-silent-subprocess) in my repo, but it's against #2112. I'll submit a separate PR after #2112 is looked at; if #2112 won't be merged, I can extract the patch and apply it against `master` for the new PR.

**Comment by hyPiRion** [comment]
Hmm, yeah, that's counterintuitive. Happy to take that patch, but you can assume #2112 will be merged in after I've reviewed it.
